pkg/pushsync/pushsync.go

```diff
@@ -352,86 +336,108 @@ func (ps *PushSync) pushToClosest(ctx context.Context, ch swarm.Chunk) (rr *pb.R
 			// if ErrWantSelf is returned, it means we are the closest peer.
 			return nil, fmt.Errorf("closest peer: %w", err)
 		}
-
-		// save found peer (to be skipped if there is some error with him)
 		skipPeers = append(skipPeers, peer)
 
-		deferFuncs = append(deferFuncs, func() {
-			if lastErr != nil {
-				ps.metrics.TotalErrors.Inc()
-				logger.Errorf("pushsync: %v", lastErr)
-			}
-		})
+		go func(peer swarm.Address, ch swarm.Chunk) {
+			ctxd, canceld := context.WithTimeout(ctx, timeToLive)
+			defer canceld()
 
-		// compute the price we pay for this receipt and reserve it for the rest of this function
-		receiptPrice := ps.pricer.PeerPrice(peer, ch.Address())
+			r, attempted, err := ps.pushPeer(ctxd, peer, ch)
+			// attempted is true if we get past accounting and actually attempt
+			// to send the request to the peer. If we dont get past accounting, we
+			// should not count the retry and try with a different peer again
+			if attempted {
+				allowedRetries--
+			}
+			if err != nil {
+				logger.Debugf("could not push to peer %s: %v", peer.String(), err)
+				resultC <- &pushResult{err: err}
```

Comment:
this error does not seem to be read anywhere. 
Consider using the errgroup package.